### ISSUE TITLE
Prüfscripte für WLAN-Mesh-Verlust migrieren

### ISSUE THREAD

**Issue by RubenKelevra** [VfN-NRW/offline-ssid#6]
Adorfer sprach an, dass es Probleme mit den WLAN-Treibern gibt und er die Check-Scripte gern mit in diesem Repo hätte.

**Comment by Adorfer** [comment]
Leider(!) war der Ansatz der "totes wifi erkennen" NICHT erfolgreich, 
mittels "iw scan dev mesh0 >/dev/nul &" und dann warten, ob der geforkte Thread sich reglulär beendet (oder hängt)
Zumindest bei WR841v9 und CPE210v1. 

Mein Vorschlag: 

alle 5 Minuten: syslog enthält "ath: phy[.]: Failed to stop [.]DMA"
-> sleep 30s -> /sbin/wifi

alle h:: Gerät hat keine AP-Clients und keine Wifimesh-Partner seit >1h (keine Ahnung, wie man das ermittelt)-> /sbin/wifi aufrufen..

alle 6h:  Gerät hatte keine AP-Clients und keine WifimeshPartner seit 24h-> reboot. 

Ach ja, die ganze Zeitautomatiken sollten davor gefeit sein, wenn ein Knoten keine korrekte Zeit hat oder die Zeit (wenn es wieder Netz gibt) Sprünge macht. Unlogisch hohe Zeitabstände sollten nicht zu einem sofortigen Reboot führen.

Ja, ich weiss, es wird langsam komplex. Aber es wäre mir wirklich etwas wert.

**Comment by RubenKelevra** [comment]
/sbin/wifi löst meiner Erfahrung nach leider das Problem in den seltensten Fällen.

Das Failed to stop DMA parsen bringt leider nichts mehr, der Bug bei dem das WLAN ziehmlich sicher hängt wird schon von einem Gluon-Script abgefangen, die Meldung hat eher verbose-Charakter ohne das damit auf wirkliche Kommunikationsprobleme geschlossen werden kann. 

Wir haben folgende Scripte im Einsatz:

Wir pingen eine Anycast-Adresse der Gateways, da sich der IPv6-Stack aufzuhängen beliebt, wenn das zwei mal I'm Abstand von 5 Minuten fehl schlägt rebooten wir.

Hängt iw, sprich die Kernel-Schnittstelle zum WLAN, kommt ein einfaches "iw" nicht mehr zurück, daraufhin rebooten wir sofort.

Wenn kein Batman-Adv-Gateway sichtbar ist, zwei mal im Abstand von 5 Minuten rebooten wir.

Wenn Mesh-Nachbarn nach dem Boot erreichbar waren, aber die alle verloren gehen rebooten wir, wenn das zweimal im Abstand von 5 Minuten der Fall ist.

Wir rebooten jeden Morgen um 5 Uhr, wenn keine Clients im WLAN sind und jeden Montag Morgen um 5 Uhr auf jeden Fall. So können wir die letzten WLAN-Probleme wenigstens innerhalb von 24h erschlagen.

**Comment by Adorfer** [comment]
Danke für die Erläuterungen. Dass da ein gerütteter Haufen von Gehirnschmalz drin steckt, das hatte ich schon gesehen. Dass es aber so viel Logik ist: Alle Achtung. 

Ich überlege noch, ob das Script sicher gegen (realischer Weise mögliche) Zeitsprünge ist, z.B bei erst später nach dem Boot erreichbaren NTP-Servern. 

Als Feature-Request (wenn man sowieso schon so eine ausgebuffte "bin ich Robinson"-Logik hat: Sinnvoll wäre eine Funktion, die bei fortwährend/fortgesetzt fehlenden Meshpartnern zumindest zeitweise mal auf anderen Kanälen nach einem bekannten Wifimesh schaut und dann ggf. dorthin wechselt bis zum nächsten Reboot. (Und dann bis dahin hoffentlich ein Update gezogen hat)

**Comment by RubenKelevra** [comment]
@Adorfer ja, Zeitsprünge sind kein Problem. Das Offline-SSID-Script funktioniert rein mit der Bootzeit, sprich einem fortlaufenden Counter in Sekunden ab boot.

Die anderen Scripte werden via Cron gestartet, die sind alle nicht wirklich hübsch ... die sind alle in unserem Statistik-Script. :)

**Comment by Adorfer** [comment]
Mit gluon 2015.2 ändern sich die Namen der WIFIs (client0) erneut, oder?
